MultiPointRasterGM.h

```diff
@@ -0,0 +1,295 @@
+                                                            MultiDbl* achievedPrecisions = NULL,
+                                                            WarningList* warnings = NULL) const = 0;
+   //> This method, for the given vector of imagePts (vector of line, sample in full image space
+   //  pixels), returns a vector of position and direction of the imaging locus
+   //  nearest the vector of given groundPts (vector of x,y,z in ECEF meters).
+   //
+   //  Note that there are two opposite directions possible.  Both are
+   //  valid, so either can be returned; the calling application can convert
+   //  to the other as necessary.
+   //
+   //  Iterative algorithms will use desiredPrecision, in meters, as the
+   //  convergence criterion for the locus position, otherwise it will be
+   //  ignored.
+   //
+   //  If a non-NULL achievedPrecision argument is received, it will be
+   //  populated with the actual precision, in meters, achieved by iterative
+   //  algorithms and 0.0 for deterministic algorithms.
+   //
+   //  If a non-NULL warnings argument is received, it will be populated
+   //  as applicable.
+   //<
+   
+   virtual MultiEcefLocus multiImageToRemoteImagingLocus(const MultiImageCoord& imagePts,
+                                                         double desiredPrecision = 0.001,
+                                                         MultiDbl* achievedPrecisions = NULL,
+                                                         WarningList* warnings = NULL) const = 0;
+   //> This method, for the given vector of imagePts (vector of line, sample in full image space
+   //  pixels), returns a vector of positions and directions of the imaging locus
+   //  at the sensor.
+   //
+   //  Note that there are two opposite directions possible.  Both are
+   //  valid, so either can be returned; the calling application can convert
+   //  to the other as necessary.
+   //
+   //  Iterative algorithms will use desiredPrecision, in meters, as the
+   //  convergence criterion for the locus position, otherwise it will be
+   //  ignored.
+   //
+   //  If a non-NULL achievedPrecision argument is received, it will be
+   //  populated with the actual precision, in meters, achieved by iterative
+   //  algorithms and 0.0 for deterministic algorithms.
+   //
+   //  If a non-NULL warnings argument is received, it will be populated
+   //  as applicable.
+   //
+   //  Notes:
+   //
+   //  The remote imaging locus is only well-defined for optical sensors.
+   //  It is undefined for SAR sensors and might not be available for
+   //  polynomial and other non-physical models.  The
+   //  imageToProximateImagingLocus method should be used instead where
+   //  possible.
+   //<
+   
+   virtual MultiEcefVector multiGetIlluminationDirection(const MultiEcefCoord& groundPts) const = 0;
+   virtual MultiDbl multiGetImageTime(const MultiImageCoord& imagePts) const = 0;
```

Comment:
Are all these multi-point accessors really that useful?  I suppose they probably aren't that hard to implement for plugins, but is there really a need to get the image times or illumination directions of lots of points in a single call?  And will that really be that much faster than doing lots of calls?  I'm worried about this interface becoming too complicated.

Reply:
these in particular were requested by JKarp.